packages/trpc/server/routers/viewer/slots/util.ts

```diff
@@ -201,6 +198,7 @@ export async function getEventType(
         },
       },
       hosts: {
+        ...(roundRobinUsernamePool ? { where: { user: { username: { in: roundRobinUsernamePool } } } } : {}),
```

Comment:
If `roundRobinUsernamePool` is passed then when fetching the event type we should also only fetch the hosts that belong in that pool of usernames. This way we don't need to fetch the availability of all the hosts.